zerver/models.py

```diff
@@ -315,11 +319,19 @@ def get_realm_by_email_domain(email):
     if settings.REALMS_HAVE_SUBDOMAINS:
         raise GetRealmByDomainException(
             "Cannot get realm from email domain when settings.REALMS_HAVE_SUBDOMAINS = True")
-    try:
-        alias = RealmAlias.objects.select_related('realm').get(domain = email_to_domain(email))
+    domain = email_to_domain(email)
+    query = RealmAlias.objects.select_related('realm')
+    alias = query.filter(domain=domain).first()
+    if alias is not None:
         return alias.realm
-    except RealmAlias.DoesNotExist:
-        return None
+    else:
```

Comment:
Does it make more sense here for us to just get all the RealmAlias objects up front and then just iterate through them to see if any of them make for a match?